app/scripts/migrations/122.ts

```diff
@@ -0,0 +1,53 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty } from '@metamask/utils';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 122;
+
+/**
+ * This migration sets the preference `isConfirmationAdvancedDetailsOpen` to
+ * `true` if the user has enabled `useNonceField` or `sendHexData`.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly
+ * what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by
+ * controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function transformState(state: Record<string, any>) {
+  if (
+    !hasProperty(state, 'PreferencesController') ||
```

Comment:
If we need it below, could we assign this and check it with a standard falsey check?

```
const preferencesState = state?.PreferencesController;

if(!preferencesState) {
    return state;
}
```